### PR TITLE
Camera2 support

### PR DESCRIPTION
**Pull request by itome**
## Changes
・Add support for camera 2 API
・Use Camera2 API if SDK_INT > 21 in sample project

## Screenshot
Android 4.3↓
ezgif-3-37aca198909a
Android 9.0↓
ezgif-3-5cb8f215f1b1

### CODE REVIEW

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt [hunk 1]
```diff
@@ -65,18 +62,19 @@ class CameraActivity : AppCompatActivity() {
...
         }
         findViewById<View>(R.id.img_switch_camera).run {
             setOnClickListener { cameraLoader.switchCamera() }
-            if (!cameraHelper.hasFrontCamera() || !cameraHelper.hasBackCamera()) {
+            if (!cameraLoader.hasMultipleCamera()) {
                 visibility = View.GONE
             }
         }
+        cameraLoader.setOnPreviewFrameListener { data, width, height ->
+            gpuImageView.gpuImage.updatePreviewFrame(data, width, height)
```
Comment (wasabeef) [inline sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt]: I'd like to use this method more helpful, so could you make define to GPUImageView too and call that?

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        return Camera.getNumberOfCameras() > 1
+    }
+
+    private fun setUpCamera() {
+        val id = getCurrentCameraId()
+        cameraInstance = getCameraInstance(id)
+        if (cameraInstance == null) {
+            Log.e("Camera1Loader", "Camera not found")
+        }
+        val parameters = cameraInstance!!.parameters
```
Comment (wasabeef) [inline sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt]: When the `cameraInstance` is null, it should quit this setup.

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt [hunk 3]
```diff
@@ -65,18 +62,19 @@ class CameraActivity : AppCompatActivity() {
...
+        }
+        gpuImageView.setRotation(getRotation(cameraLoader.getCameraOrientation()))
```
Comment (wasabeef) [inline sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt]: Each time switch to the camera, we have to check camera rotation.

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            }
+        }
+        return 0
+    }
+
+    private fun getCameraInstance(id: Int): Camera? {
+        return try {
+            Camera.open(id)
+        } catch (e: Exception) {
+            Log.e("Camera not found", e.toString())
```
Comment (wasabeef) [inline sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt]: Or you can throw to like `IllegalAccessError("Camera not found")`, and in that case the `cameraInstance` be lateinit.

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt (hunk 1 above)
Comment (itome) [reply sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt]: Done!

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt (hunk 3 above)
Comment (itome) [reply sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/activity/CameraActivity.kt]: Done!

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt (hunk 4 above)
Comment (itome) [reply sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt]: Done

File: sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt (hunk 2 above)
Comment (itome) [reply sample/src/main/java/jp/co/cyberagent/android/gpuimage/sample/utils/Camera1Loader.kt]: Done

**Pull request merged**